test/functional/full_text_search/search_controller_test.rb

```diff
@@ -538,10 +538,14 @@ def test_search
         search("redmine")
         revision10 = @repository.changesets.find_by(revision: "10").filechanges
         revision11 = @repository.changesets.find_by(revision: "11").filechanges
+        revision14 = @repository.changesets.find_by(revision: "14")&.filechanges
```

Comment:
Redmine 4.2 don't have '@repository.changesets.find_by(revision: "14")'.
We can check Redmine::Version in stead of checking the result of '@repository.changesets.find_by(revision: "14")' is `nil`.

Reply:
I prefer to the following style:

```ruby
changeset14 = @repository.changesets.find_by(revision: "14")
# ...
items = []
if changeset14
  items << changeset14.filechanges.find_by(path: ...)
end
items << revision11.find_by(...)
items << revision10.find_by(...)
```